### PR TITLE
separate out app logic from rollup logic

### PR DESCRIPTION
**Pull request by lobstergrindset**
Refactored code to have application specific logic separate from the basic logic to drive the rollup forward. Was doing this for a defi rollup I was working on and thought others would benefit from the separation too.

Main changes:
- Moved all rollup application logic into new file `messenger_logic.go`
- Made rest of codebase operate on `[]byte` instead of `Transactions` to make easier to write new rollups
- Renamed files to reflect new logic contained in them
    - messenger.go -> rollup_app.go (has logic for rollup app that talks to web clients)
    - rollup.go -> rollup_blocks.go (only has logic relevant to managing blocks for a rollup)
    - messenger_logic.go : new file holding logic specific to the messenger rollup

### CODE REVIEW

File: messenger/messenger_logic.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	a.restRouter.HandleFunc("/message", a.postMessage).Methods("POST")
+	a.restRouter.HandleFunc("/recent", a.getRecentMessages).Methods("GET")
+}
+
+// encode transaction into bytes to be sent to the sequencer
+func encodeTx(tx Transaction) ([]byte, error) {
+	data, err := json.Marshal(tx)
+	if err != nil {
+		log.Errorf("error encoding transaction: %s\n", err)
+		return data, err
```
Comment (noot) [inline messenger/messenger_logic.go]:
```suggestion
		return nil, err
```

File: messenger/messenger_logic.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+// decode transaction from bytes back into rollup format
+func decodeTx(txEncoded []byte) (Transaction, error) {
+	tx := &Transaction{}
+	if err := json.Unmarshal(txEncoded, tx); err != nil {
+		log.Errorf("error decoding transaction: %s\n", err)
+		return *tx, err
+	}
+	return *tx, nil
+}
```
Comment (noot) [inline messenger/messenger_logic.go]:
```suggestion
func decodeTx(txEncoded []byte) (*Transaction, error) {
	tx := &Transaction{}
	if err := json.Unmarshal(txEncoded, tx); err != nil {
		log.Errorf("error decoding transaction: %s\n", err)
		return nil, err
	}
	return tx, nil
}
```

File: messenger/messenger_logic.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	var tx Transaction
+	// decode transaction to ensure proper format
+	err := json.NewDecoder(r.Body).Decode(&tx)
+	if err != nil {
+		log.Errorf("error decoding transaction: %s\n", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// recode transaction to send to sequencer
```
Comment (noot) [inline messenger/messenger_logic.go:67]:
```suggestion
	// re-encode transaction to send to sequencer
```

**Review by itamarreif — APPROVED** [review]
Besides what @noot mentioned about return values, seems fine and shuoldn't change the functionality. If it helps with developing other rollups, then should be good to merge.

**Review by mycodecrafting — APPROVED** [review]
let's get this modular 🦪 !

**Pull request merged**